eyelid-match-ops/src/primitives/poly.rs

```diff
@@ -60,65 +60,191 @@ pub fn cyclotomic_mul(a: &Poly, b: &Poly) -> Poly {
 /// The returned polynomial has maximum degree [`MAX_POLY_DEGREE`].
 pub fn karatsuba_mul(a: &Poly, b: &Poly) -> Poly {
     let mut res;
-    let n = a.degree() + 1;
+    let n = a.degree() + 1; // invariant: n is a power of 2
+    debug_assert!(n.count_ones() == 1); // checking the invariant
```

Comment:
Yes, indeed, and it could lead to problems. Maybe we should have a unit test forcing such scenario. Then I guess we will have to compute the "next power of 2 that is larger than `a.degree() + 1`".

Reply:
I think it's very likely in a random polynomial, just not guaranteed.

We could drop the assert for now, but add a TODO or open a ticket?

Reply:
So for now let's go with:
```suggestion
    // invariant: n is a power of 2, before and after this function runs
    static_assert!(MAX_POLY_DEGREE.count_ones() == 1);
    static_assert!(MIN_KARATSUBA_REC_DEGREE.count_ones() == 1);
```

Reply:
I also added a similar invariant to `flat_karatsuba_mul()`.